### DISCUSSION TITLE
Add a dark block background to tables when rendering markdown in React

### DISCUSSION THREAD

**Discussion by chengtie** [remarkjs/remark#1121] [Q&A] [answered]
Hello,

I have followed [this thread](https://designly.biz/blog/post/react-markdown-how-to-create-a-copy-code-button) to add a dark block background for code when rendering markdown in React. Here is the code and codesandbox: https://codesandbox.io/s/happy-flower-2g42h4?file=/src/PostBody.tsx

```
import React from "react";
import ReactMarkdown from "react-markdown";
import rehypeRaw from "rehype-raw";
import remarkGfm from "remark-gfm";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { a11yDark } from "react-syntax-highlighter/dist/cjs/styles/prism";

export default function PostBody({ content }) {
  // Add the CodeCopyBtn component to our PRE element
  const Pre = ({ children }) => {
    return <pre className="blog-pre">{children}</pre>;
  };

  return (
    <ReactMarkdown
      className="post-markdown"
      linkTarget="_blank"
      rehypePlugins={[rehypeRaw]}
      remarkPlugins={[remarkGfm]}
      components={{
        pre: Pre,
        code({ node, inline, className = "blog-code", children, ...props }) {
          const match = /language-(\w+)/.exec(className || "");
          return !inline && match ? (
            <SyntaxHighlighter
              style={a11yDark}
              language={match[1]}
              PreTag="div"
              {...props}
            >
              {String(children).replace(/\n$/, "")}
            </SyntaxHighlighter>
          ) : (
            <code className={className} {...props}>
              {children}
            </code>
          );
        }
      }}
    >
      {content}
    </ReactMarkdown>
  );
}
```

Now, I would like to also add a dark block background to tables in the markdown. A table is in general like `\n | X | Y | \n |-------|-------| \n | 1 | 4 | \n | 2 | 3 | \n` as a markdown string. It seems that [remark-gfm](https://github.com/remarkjs/remark-gfm) supports tables, but I don't know how to use it.

Could anyone help?

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @chengtie!
You can add backgrounds and styles using CSS.
There are some wonderful guides for styling tables with CSS, like: https://www.w3schools.com/Css/css_table.asp, https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Styling_tables, and https://www.digitalocean.com/community/tutorials/how-to-style-a-table-with-css

If you're looking for some general dark mode styles, similar to those of GitHub, check out https://github.com/sindresorhus/github-markdown-css

**Reply by chengtie** [reply]
Thank you, Christian,

I'm looking for a non-CSS way as what `PostBody` does for code blocks. I plan to do more things than adding a dark background, e.g., adding a copy button. So I would like to be able to identify the tables and manipulate them.

Hope my request is clearer now.

**Reply by ChristianMurphy** [reply]
The `components` prop allows you to render tables with any custom or community component you'd like. https://github.com/remarkjs/react-markdown#props

Check out https://github.com/brillout/awesome-react-components#editable-data-grid--spreadsheet and https://github.com/brillout/awesome-react-components#table as some starting points.

**Reply by chengtie** [reply]
> The `components` prop allows you to render tables with any custom or community component you'd like. https://github.com/remarkjs/react-markdown#props
> 
> Check out https://github.com/brillout/awesome-react-components#editable-data-grid--spreadsheet and https://github.com/brillout/awesome-react-components#table as some starting points.

I see the prop `table` which I need, thank you.
